src/parser/contexts.h

```diff
@@ -994,10 +999,20 @@ struct ParseDefsCtx : TypeParserCtx<ParseDefsCtx> {
                                        type.getSignature().results));
   }
 
-  Result<> finishBlock(Index pos, InstrsT) {
+  Result<> makeIf(Index pos, std::optional<Name> label, HeapType type) {
+    // TODO: validate labels?
+    // TODO: Move error on input types to here?
+    return withLoc(
+      pos,
+      irBuilder.makeIf(label ? *label : Name{}, type.getSignature().results));
+  }
+
+  Result<> visitEnd(Index pos, InstrsT) {
```

Comment:
same point Alon made above, move visitEnd to be after makeBlock?

Reply:
I think I ended up doing this later in this change of PRs, so I'll leave it for now to reduce churn.